bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/VoiceManager.java

```diff
@@ -469,6 +470,25 @@ public TTSService getTTS() {
     }
 
     /**
+     * Retrieves a TTS service with the given id.
+     *
+     * @param id the id of the TTS service
+     * @return a TTS service or null, if no service with this id exists
+     */
+    public TTSService getTTS(String id) {
+        return ttsServices.get(id);
+    }
+
+    /**
+     * Retrieves all TTS services.
+     *
+     * @return a collection of TTS services
+     */
+    public Collection<TTSService> getTTSs() {
+        return Collections.unmodifiableCollection(ttsServices.values());
```

Comment:
HashMap.values() returns a view of the values.
Collections.unmodifiableCollection(...) returns an unmodifiable view.

So, if you use that construct every change to e.g. ttsServices (an add or remove of a service) will be mirrored to the caller that is using the returned view (only the view is unmodifiable not the collection the view resides on).
If the caller iterates over the returned collection and ttsServices is changes, this will lead to a CME.

Same for the other functions, too.

The normal use case is to create a temp. collection to create the unmodifiable view (so the only one that holds the reference is the unmodifiable view at all).
As you need to copy the values to avoid a CME, there is perhaps no need to use unmodifiable at all and leave the ownership of the collection to the caller.